### PR TITLE
[ToricVarieties] Tests for blowup do not depend on order of rays anymore

### LINKED ISSUE #4348: Tests in `ToricVarieties/toric_blowups.jl` depend on order of `rays(normal_fan(simplex(2)))`
I have seen some test failures there today on CI, but only some of the jobs failed there, e.g. see 
- https://github.com/oscar-system/Oscar.jl/actions/runs/12012847800/job/33485041626#step:10:1174 (failure 1.11)
- https://github.com/oscar-system/Oscar.jl/actions/runs/12012847800/job/33485042881#step:10:1114 (failure nightly)
- https://github.com/oscar-system/Oscar.jl/actions/runs/12012847800/job/33485044769#step:10:4079 (success 1.10-macOS)

Do the tests here or the functions they test employ any kind of randomness? If yes, maybe the tests need to be adapted a bit like e.g. in https://github.com/oscar-system/Oscar.jl/pull/4153 by @benlorenz .

failure log:
```julia
Basic tests for simple toric blowup: Test Failed at /home/runner/work/Oscar.jl/Oscar.jl/test/AlgebraicGeometry/ToricVarieties/toric_blowups.jl:44
  Expression: matrix(morphism_on_torusinvariant_weil_divisor_group(bl)) == matrix(ZZ, [1 0 0; 0 1 0; 0 0 1; 1 1 0])
   Evaluated: [1 0 0; 0 1 0; 0 0 1; 1 0 1] == [1 0 0; 0 1 0; 0 0 1; 1 1 0]

Stacktrace:
 [1] macro expansion
   @ /opt/hostedtoolcache/julia/1.11-nightly/x64/share/julia/stdlib/v1.11/Test/src/Test.jl:679 [inlined]
 [2] macro expansion
   @ ~/work/Oscar.jl/Oscar.jl/test/AlgebraicGeometry/ToricVarieties/toric_blowups.jl:44 [inlined]
 [3] macro expansion
   @ /opt/hostedtoolcache/julia/1.11-nightly/x64/share/julia/stdlib/v1.11/Test/src/Test.jl:1704 [inlined]
 [4] macro expansion
   @ ~/work/Oscar.jl/Oscar.jl/test/AlgebraicGeometry/ToricVarieties/toric_blowups.jl:42 [inlined]
 [5] macro expansion
   @ /opt/hostedtoolcache/julia/1.11-nightly/x64/share/julia/stdlib/v1.11/Test/src/Test.jl:1704 [inlined]
 [6] top-level scope
   @ ~/work/Oscar.jl/Oscar.jl/test/AlgebraicGeometry/ToricVarieties/toric_blowups.jl:3

GC: pause 284.08ms. collected 555.574379MB. incr 
Heap stats: bytes_mapped 1280.31 MB, bytes_resident 998.86 MB,
heap_size 1346.21 MB, heap_target 1429.95 MB, Fragmentation 0.563
Properties of toric blowup defined by ideal: Test Failed at /home/runner/work/Oscar.jl/Oscar.jl/test/AlgebraicGeometry/ToricVarieties/toric_blowups.jl:76
  Expression: II == center_unnormalized(bl)
   Evaluated: Sheaf of ideals on normal, smooth toric variety without torusfactor == Sheaf of ideals on normal, smooth toric variety without torusfactor

Stacktrace:
 [1] macro expansion
   @ /opt/hostedtoolcache/julia/1.11-nightly/x64/share/julia/stdlib/v1.11/Test/src/Test.jl:679 [inlined]
 [2] macro expansion
   @ ~/work/Oscar.jl/Oscar.jl/test/AlgebraicGeometry/ToricVarieties/toric_blowups.jl:76 [inlined]
 [3] macro expansion
   @ /opt/hostedtoolcache/julia/1.11-nightly/x64/share/julia/stdlib/v1.11/Test/src/Test.jl:1704 [inlined]
 [4] macro expansion
   @ ~/work/Oscar.jl/Oscar.jl/test/AlgebraicGeometry/ToricVarieties/toric_blowups.jl:76 [inlined]
 [5] macro expansion
   @ /opt/hostedtoolcache/julia/1.11-nightly/x64/share/julia/stdlib/v1.11/Test/src/Test.jl:1704 [inlined]
 [6] top-level scope
   @ ~/work/Oscar.jl/Oscar.jl/test/AlgebraicGeometry/ToricVarieties/toric_blowups.jl:3
Test Summary:                                                             | Pass  Fail  Total   Time
Toric blowups                                                             |   31     2     33  24.2s
  Basic properties of BP2                                                 |    9            9   0.3s
  Basic attributes of BP2                                                 |    7            7   0.3s
  Trigger issue of PR3006                                                 |    2            2   0.6s
  Basic tests for simple toric blowup                                     |    3     1      4   2.4s
  Arithmetics, comparison and composition                                 |    4            4   0.3s
  Strict, total transform and exceptional divisor for simple toric blowup |    4            4   6.0s
  Properties of toric blowup defined by ideal                             |    2     1      3   2.2s
ERROR: LoadError: LoadError: Some tests did not pass: 31 passed, 2 failed, 0 errored, 0 broken.
in expression starting at /home/runner/work/Oscar.jl/Oscar.jl/test/AlgebraicGeometry/ToricVarieties/toric_blowups.jl:1
```
pinging @HereAround since this is about toric varieties

### PR DESCRIPTION
**Pull request by HereAround**
This PR aims to close https://github.com/oscar-system/Oscar.jl/issues/4348.

cc @lgoettgens @lkastner

### CODE REVIEW

File: test/AlgebraicGeometry/ToricVarieties/toric_blowups.jl [hunk 1]
```diff
@@ -31,8 +31,8 @@
   amb1 = domain(bd1)
   
   @testset "Trigger issue of PR3006" begin
-    @test is_complete(amb1)
-    @test is_simplicial(amb1)
+    @test is_complete(amb1) == true
+    @test is_simplicial(amb1) == true
```
Comment (lgoettgens) [inline test/AlgebraicGeometry/ToricVarieties/toric_blowups.jl]:
```suggestion
    @test is_complete(amb1)
    @test is_simplicial(amb1)
```
I am not a fan of all of these `== true` here and further below

**Comment by lgoettgens** [comment]
I would leave the question if that is mathematically sensible here to @lkastner and @benlorenz , since I do not really now whats going on here.

File: test/AlgebraicGeometry/ToricVarieties/toric_blowups.jl (hunk 1 above)
Comment (HereAround) [reply test/AlgebraicGeometry/ToricVarieties/toric_blowups.jl]: Yeah, I removed those just now.

**Comment by HereAround** [comment]
> I would leave the question if that is mathematically sensible here to @lkastner and @benlorenz , since I do not really now whats going on here.

Mathematically, the only significant change is that I am asking for the rank of a matrix instead of comparing it against an explicit hard-coded form of that matrix (which depends on the order of the rays).

But then, fair.

**Comment by benlorenz** [comment]
In the log in #4348 there is a second failure which is probably related:
```
Properties of toric blowup defined by ideal: Test Failed at /home/runner/work/Oscar.jl/Oscar.jl/test/AlgebraicGeometry/ToricVarieties/toric_blowups.jl:76
  Expression: II == center_unnormalized(bl)
   Evaluated: Sheaf of ideals on normal, smooth toric variety without torusfactor == Sheaf of ideals on normal, smooth toric variety without torusfactor
```

**Review by lkastner — APPROVED** [review]

**Pull request merged**